docs/proposals/caching/README.md

```diff
@@ -0,0 +1,254 @@
+- Accelerate processing speed of resource requests across regions.
+- Get resources in multiple clusters by labels.
+
+## Design Details
+
+### Define the scope of the cached resource
+
+#### New ClusterCache API
+
+We propose a new CR in `cluster.karmada.io` group.
+
+```golang
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// ClusterCache is the Schema for the cluster cache API
+type ClusterCache struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterCacheSpec   `json:"spec,omitempty"`
+	Status ClusterCacheStatus `json:"status,omitempty"`
+}
+
+// ClusterCacheSpec defines the desired state of ClusterCache
+type ClusterCacheSpec struct {
+	// ClusterSelectors represents the filter to select clusters.
+	// +required
+	ClusterSelectors []ClusterSelector `json:"clusterSelectors"`
+
+	// ResourceSelectors used to select resources.
+	// +required
+	ResourceSelectors []ResourceSelector `json:"resourceSelectors"`
+
+	// StatusUpdatePeriodSeconds is the period to update the status of the resource.
+	// default is 10s.
+	// +optional
+	StatusUpdatePeriodSeconds uint32 `json:"statusUpdatePeriodSeconds,omitempty"`
+}
+
+// ClusterAffinity represents the filter to select clusters.
+type ClusterSelector struct {
+	// LabelSelector is a filter to select member clusters by labels.
+	// If non-nil and non-empty, only the clusters match this filter will be selected.
+	// +optional
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+
+	// FieldSelector is a filter to select member clusters by fields.
+	// If non-nil and non-empty, only the clusters match this filter will be selected.
+	// +optional
+	FieldSelector *FieldSelector `json:"fieldSelector,omitempty"`
+
+	// ClusterNames is the list of clusters to be selected.
+	// +optional
+	ClusterNames []string `json:"clusterNames,omitempty"`
+
+	// ExcludedClusters is the list of clusters to be ignored.
+	// +optional
+	ExcludeClusters []string `json:"exclude,omitempty"`
+}
+
+// FieldSelector is a field filter.
+type FieldSelector struct {
+	// A list of field selector requirements.
+	MatchExpressions []corev1.NodeSelectorRequirement `json:"matchExpressions,omitempty"`
+}
+
+// ResourceSelector the resources will be selected.
+type ResourceSelector struct {
+	// APIVersion represents the API version of the target resources.
+	// +required
+	APIVersion string `json:"apiVersion"`
+
+	// Kind represents the Kind of the target resources.
+	// +required
+	Kind string `json:"kind"`
+
+	// Namespace of the target resource.
+	// Default is empty, which means all namespaces.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// Indexes is a list of custom index
+	// Default is empty, which means no custom index.
+	// +optional
+	Indexes []ResourceIndexRef `json:"indexes,omitempty"`
```

Comment:
A question came to my mind: why user cares about the index thing? Does the cache( the informer cache) intend exposed to the user directly?

Reply:
I agree, the ability provided by the cache does not need the user to care, I will remove index.

Reply:
What about remove it?

Reply:
+1
If the user doesn't care about it, then it should be removed from API.

Reply:
Removed it.